### PR TITLE
Arkcase Dockerimage

### PR DESCRIPTION
**Pull request by naveen-armedia**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1 +1,79 @@
 FROM 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_base_java8:latest
+LABEL ORG="Armedia LLC" \
+      VERSION="1.0" \
+      IMAGE_SOURCE=https://github.com/ArkCase/ark_arkcase_core \
+      MAINTAINER="Armedia LLC"
+
+ARG ARKCASE_VERSION=2021.03-RC11
+ARG TOMCAT_VERSION=9.0.50 
+ARG TOMCAT_MAJOR_VERSION=9
+ARG SYMMENTRIC_KEY=9999999999999999999999
```
Comment (armedia-customer-service) [inline Dockerfile]: typo, should be `SYMMETRIC_KEY`

Comment (armedia-customer-service) [reply Dockerfile]: Can you take all these version numbers from the environment somehow or do they have to be hardcoded in the Dockerfile?

**Review by dzdimov — DISMISSED** [review]

File: Dockerfile (hunk 1 above)
Comment (naveen-armedia) [reply Dockerfile]: removed hardcoding of version details in the dockerfile.

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          ECR_REPOSITORY: arkcase_core
+          TOMCAT_VERSION: 9.0.50 
+          TOMCAT_MAJOR_VERSION: 9
+          SYMMETRIC_KEY: 99999999
+          MARIADB_CONNECTOR_VERSION: 2.2.5
+          RESOURCE_PATH: artifacts
+        run: |
+          docker build -t  $ECR_REGISTRY/$ECR_REPOSITORY:$ARKCASE_VERSION --build-arg TOMCAT_VERSION --build-arg  TOMCAT_MAJOR_VERSION --build-arg  SYMMETRIC_KEY --build-arg MARIADB_CONNECTOR_VERSION --build-arg RESOURCE_PATH .
```
Comment (david-oc-miller) [inline .github/workflows/main.yaml]: I see the `Dockerfile` still has equivalent `ARG` elements.  Do these environment variables and build args take precedence over the ARGs in the Dockerfile?

Comment (naveen-armedia) [reply .github/workflows/main.yaml]: Yes @david-oc-miller

Comment (david-oc-miller) [reply .github/workflows/main.yaml]: OK thanks, I see how it works from here: https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables---build-arg.

I do notice that you are not passing ARKCASE_VERSION as a build-arg?

Comment (naveen-armedia) [reply .github/workflows/main.yaml]: Thanks for the notice. Added ARKCASE_VERSION now

**Review by dzdimov — DISMISSED** [review]

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: download artifacts
+        run: ./getArtifact.sh
+        env:
+          ARKCASE_VERSION: 2021.03-RC11
```
Comment (bojmil-armedia) [inline .github/workflows/main.yaml]: Isn't ARKCASE_VERSION defined on line 10?

File: getArtifact.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+here=$(realpath "$0")
+here=$(dirname "$here")
+cd "$here"
+rm -rf artifacts
+mkdir artifacts
+echo "Downloading Arkcase core artifacts $ARKCASE_VERSION"
+aws s3 cp "s3://arkcase-container-artifacts/arkcase-core/${ARKCASE_VERSION}/artifacts/" artifacts/ --recursive
```
Comment (bojmil-armedia) [inline getArtifact.sh:8]: What artifacts are we downloading here? Is it only ArkCase artifacts or also the dependencies artifacts (Alfresco, Solr, ActiveMQ,...)?  
All artifacts could be downloaded from their release site. For ArkCase it's GitHub. The other dependencies have their own sites.

File: Dockerfile [hunk 5]
```diff
@@ -1 +1,79 @@
 FROM 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_base_java8:latest
+LABEL ORG="Armedia LLC" \
+      VERSION="1.0" \
```
Comment (bojmil-armedia) [inline Dockerfile:3]: This should be ARKCASE_VERSION

File: Dockerfile [hunk 6]
```diff
@@ -1 +1,79 @@
...
+    yum clean all
+    
+RUN yum -y install epel-release epel-testing && \
+    yum install -y tesseract tesseract-osd qpdf ImageMagick ImageMagick-devel && \
+    ln -s /usr/bin/convert /usr/bin/magick &&\
+    ln -s /usr/share/tesseract/tessdata/configs/pdf /usr/share/tesseract/tessdata/configs/PDF &&\
+    yum clean all
+USER tomcat
+
+EXPOSE 8005
```
Comment (bojmil-armedia) [inline Dockerfile:77]: Why 8005? Is tomcat running on this port? Why not 443?

File: getArtifact.sh (hunk 4 above)
Comment (naveen-armedia) [reply getArtifact.sh:8]: only Arkcase application war file and mariadb sql driver.

Comment (bojmil-armedia) [reply getArtifact.sh:8]: Then we need to define where you can find all ArkCase releases. We have final releases in GitHub https://github.com/ArkCase/ArkCase/releases. But we don't add release candidates, snapshots and feature builds.

File: .github/workflows/main.yaml (hunk 3 above)
Comment (david-oc-miller) [reply .github/workflows/main.yaml]: @naveen-armedia , do we need `ARKCASE_VERSION` at line 10 and also at line 22?  Or, is the `env` in the steps is different from the top-level `env`?

File: .github/workflows/main.yaml (hunk 2 above)
Comment (david-oc-miller) [reply .github/workflows/main.yaml]: @naveen-armedia I still don't see where you are passing `ARKCASE_VERSION` in a `build-arg`.  Do you need to push another commit?

File: Dockerfile [hunk 7]
```diff
@@ -1 +1,79 @@
 FROM 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_base_java8:latest
+LABEL ORG="Armedia LLC" \
```
Comment (david-oc-miller) [inline Dockerfile:2]: @naveen-armedia I think we said we would not maintain e.g. `ark_base_java8` but instead add Java right into the images that need it?  I still prefer a more layered set of images, but I believe the team ended up disagreeing with me.  So we would add Java 8 as a layer in this image instead of starting from `ark_base_java8`.  Or would that be another MR?

**Review by david-oc-miller — APPROVED** [review]

**Comment by david-oc-miller** [comment]
@bojmil-armedia can you review again?  With another approval we can accept this one.

**Review by bojmil-armedia — APPROVED** [review]

File: Dockerfile [hunk 8]
```diff
@@ -1 +1,79 @@
...
+RUN tar -xf apache-tomcat-${TOMCAT_VERSION}.tar.gz && \
+    mv apache-tomcat-${TOMCAT_VERSION} tomcat && \
+    rm apache-tomcat-${TOMCAT_VERSION}.tar.gz &&\
+    # Removal of default/unwanted Applications
+    rm -rf tomcat/webapps/* tomcat/temp/* tomcat/bin/*.bat && \
+    mv server.xml logging.properties tomcat/conf/ && \
+    mkdir -p /tomcat/logs &&\
+    mv arkcase-${ARKCASE_VERSION}.war  ./tomcat/webapps/arkcase.war && \
+    # wget --directory-prefix=./tomcat/webapps/ -O arkcase.war https://github.com/ArkCase/ArkCase/releases/download/${ARKCASE_VERSION}/arkcase-${ARKCASE_VERSION}.war &&\
+    # mv arkcase.war /app/tomcat/ &&\
```
Comment (dzdimov) [inline Dockerfile:53]: We can delete these comments

Comment (bojmil-armedia) [reply Dockerfile:53]: In a next PR :)
